### ISSUE TITLE
Outline incompatibilities introduced later in the pipeline are not fatal

### ISSUE THREAD

**Issue by Hoolean** [googlefonts/fontmake#1091]
Earlier we encountered #897 as cu2qu raised a fatal error, however, when debugging the issue we tested with and without cu2qu and noticed that when cu2qu is disabled then the font build completes with an incomplete `gvar` table.

This happens as varLib skips glyphs with incompatible outlines [with a warning but not a fatal error](https://github.com/fonttools/fonttools/blob/2ce45c2861cd274a265f68a3fdb6b60902c42892/Lib/fontTools/varLib/__init__.py#L334); the compatibility checker is run at the beginning of the pipeline and so also cannot detect such issues.

For projects with quadratic sources it would be nice to disable cu2qu but keep the same level of strictness. Should we make the above warning fatal by default (perhaps only for fontmake)?

**Comment by anthrotype** [comment]
yeah, not the first time that this varLib's warning-that-should-have-been-an-error let some bug slip..
Maybe we could add a new fontTools.config option to turn that into an error.
